eigenpro/solver.py

```diff
@@ -132,21 +129,18 @@ def fit(model, X, Y, x, y, device, dtype=torch.float32, kernel=None,
                 torch.cuda.empty_cache()
 
             if ( (project_counter + 1) % T == 0 or (t==len(train_dataloader)-1) ) and accumulated_gradients:
-                projection_dataset = array_dataset.ArrayDataset(model.centers[0], optimizer.grad_accumulation)
-                projection_loader = torch_data.DataLoader(
-                    projection_dataset, batch_size=model_batch_size, shuffle=True)
-                for _ in range(1):
-                    for z_batch, grad_batch, id_batch in tqdm(
-                            projection_loader, 
-                            total=len(projection_loader), 
-                            leave=False,
-                            desc="Projection"
-                        ):
-                        optimizer.step(z_batch, grad_batch, id_batch, projection=True)
-
-                update_projection = torch.cat([k.weights_project.to(device_base) for k in model.shard_kms])
-
-                model.update_by_index(torch.tensor(list(range(p))), update_projection)
+
+                if model_preconditioner is None:
+                    # model preconditioner
+                    nys_model_indices = np.random.choice(model.centers[0].shape[0], s_model, replace=False)
+                    nys = model.centers[0][nys_model_indices, :].to(device_base)
+                    model_preconditioner = pcd.Preconditioner(kernel_fn, nys, q_model)
```

Comment:
can we have a better naming for `q_model` and `s_model`?
there is no way to guess their meanings correctly

Reply:
what is your suggestion? How about nystrom_samples_model and nystrom_truncation_level_model

Reply:
how about n_pcd_nyst_samples for `s_model`
and n_pcd_eigenvals for `q_model`